scripts/change-log-generator.js

```diff
@@ -100,213 +82,31 @@ function validateReleaseBranch(releaseBranch) {
  * new changes.
  */
 function getNewChangeLogBranch(releaseBranch) {
-  var changeLogBranch =
-    CHANGE_LOG_BRANCH + releaseBranch.replace('origin/release/v', '');
-  var command = util.format(
-    "git checkout $(git show-ref --verify --quiet refs/heads/%s || echo '-b') %s",
-    changeLogBranch,
-    changeLogBranch,
-    releaseBranch
-  );
+  var changeLogBranch = getChangeLogBranch(releaseBranch);
   if (ADD_VERBOSE_LOGGING) {
     console.log('Generating change log branch.');
-    console.log(command);
-  }
-  shell.exec(command);
-}
-
-/**
- * This command will list all commits that are different between
- * the two branches. Therefore, we are guaranteed to get all new
- * commits relevant only to the new branch.
- */
-function getCommits(releaseBranch, previousBranch) {
```

Comment:
Moved all parsing logic into a new script to reduce complexity and make this easier to read.